### ISSUE TITLE
Fails to build with Boost 1.72.0

### ISSUE THREAD

**Issue by jbeich** [mapnik/mapnik#4095]
Regressed by boostorg/gil@5611bd580718. From [error log](https://github.com/mapnik/mapnik/files/3956356/mapnik-3.0.12.1075.log):
```c++
In file included from deps/agg/src/agg_pixfmt_rgba.cpp:11:
deps/boost/gil/extension/toolbox/hsv.hpp:43:9: error: no template named 'vector3' in namespace 'boost::mpl'; did you mean 'std::vector'?
typedef mpl::vector3< hsv_color_space::hue_t
        ^~~~~~~~~~~~
        std::vector
/usr/include/c++/v1/vector:470:28: note: 'std::vector' declared here
class _LIBCPP_TEMPLATE_VIS vector
                           ^
In file included from deps/agg/src/agg_pixfmt_rgba.cpp:11:
deps/boost/gil/extension/toolbox/hsv.hpp:43:14: error: too many template arguments for class template 'vector'
typedef mpl::vector3< hsv_color_space::hue_t
             ^
/usr/include/c++/v1/vector:470:28: note: template is declared here
class _LIBCPP_TEMPLATE_VIS vector
                           ^
In file included from deps/agg/src/agg_pixfmt_rgba.cpp:7:
In file included from /usr/local/include/boost/gil.hpp:12:
In file included from /usr/local/include/boost/gil/algorithm.hpp:11:
In file included from /usr/local/include/boost/gil/bit_aligned_pixel_iterator.hpp:11:
In file included from /usr/local/include/boost/gil/bit_aligned_pixel_reference.hpp:12:
In file included from /usr/local/include/boost/gil/pixel.hpp:12:
In file included from /usr/local/include/boost/gil/channel.hpp:11:
In file included from /usr/local/include/boost/gil/utilities.hpp:11:
In file included from /usr/local/include/boost/gil/detail/mp11.hpp:12:
In file included from /usr/local/include/boost/mp11.hpp:11:
/usr/local/include/boost/mp11/list.hpp:45:19: error: no type named 'type' in 'boost::mp11::detail::mp_size_impl<int>'
template<class L> using mp_size = typename detail::mp_size_impl<L>::type;
                  ^~~~~
/usr/local/include/boost/gil/utilities.hpp:263:43: note: in instantiation of template type alias 'mp_size' requested here
        std::integral_constant<int, mp11::mp_size<ColorSpace>::value>
                                          ^
deps/boost/gil/extension/toolbox/hsv.hpp:49:9: note: in instantiation of default argument for 'layout<int>' required here
typedef layout<hsv_t> hsv_layout_t;
        ^~~~~~~~~~~~~
In file included from deps/agg/src/agg_pixfmt_rgba.cpp:12:
deps/boost/gil/extension/toolbox/hsl.hpp:43:9: error: no template named 'vector3' in namespace 'boost::mpl'; did you mean 'std::vector'?
typedef mpl::vector3< hsl_color_space::hue_t
        ^~~~~~~~~~~~
        std::vector
/usr/include/c++/v1/vector:470:28: note: 'std::vector' declared here
class _LIBCPP_TEMPLATE_VIS vector
                           ^
In file included from deps/agg/src/agg_pixfmt_rgba.cpp:12:
deps/boost/gil/extension/toolbox/hsl.hpp:43:14: error: too many template arguments for class template 'vector'
typedef mpl::vector3< hsl_color_space::hue_t
             ^
/usr/include/c++/v1/vector:470:28: note: template is declared here
class _LIBCPP_TEMPLATE_VIS vector
                           ^
In file included from deps/agg/src/agg_pixfmt_rgba.cpp:7:
In file included from /usr/local/include/boost/gil.hpp:12:
In file included from /usr/local/include/boost/gil/algorithm.hpp:11:
In file included from /usr/local/include/boost/gil/bit_aligned_pixel_iterator.hpp:11:
In file included from /usr/local/include/boost/gil/bit_aligned_pixel_reference.hpp:12:
In file included from /usr/local/include/boost/gil/pixel.hpp:12:
In file included from /usr/local/include/boost/gil/channel.hpp:11:
In file included from /usr/local/include/boost/gil/utilities.hpp:11:
In file included from /usr/local/include/boost/gil/detail/mp11.hpp:12:
In file included from /usr/local/include/boost/mp11.hpp:11:
/usr/local/include/boost/mp11/list.hpp:45:19: error: no type named 'type' in 'boost::mp11::detail::mp_size_impl<int>'
template<class L> using mp_size = typename detail::mp_size_impl<L>::type;
                  ^~~~~
/usr/local/include/boost/gil/utilities.hpp:263:43: note: in instantiation of template type alias 'mp_size' requested here
        std::integral_constant<int, mp11::mp_size<ColorSpace>::value>
                                          ^
deps/boost/gil/extension/toolbox/hsl.hpp:49:9: note: in instantiation of default argument for 'layout<int>' required here
typedef layout<hsl_t> hsl_layout_t;
        ^~~~~~~~~~~~~
deps/boost/gil/extension/toolbox/hsl.hpp:61:8: error: redefinition of 'default_color_converter_impl<boost::mp11::mp_list<boost::gil::red_t, boost::gil::green_t, boost::gil::blue_t>, int>'
struct default_color_converter_impl< rgb_t, hsl_t >
       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
deps/boost/gil/extension/toolbox/hsv.hpp:63:8: note: previous definition is here
struct default_color_converter_impl< rgb_t, hsv_t >
       ^
In file included from deps/agg/src/agg_pixfmt_rgba.cpp:12:
deps/boost/gil/extension/toolbox/hsl.hpp:163:8: error: redefinition of 'default_color_converter_impl<int, boost::mp11::mp_list<boost::gil::red_t, boost::gil::green_t, boost::gil::blue_t> >'
struct default_color_converter_impl<hsl_t,rgb_t>
       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
deps/boost/gil/extension/toolbox/hsv.hpp:144:8: note: previous definition is here
struct default_color_converter_impl<hsv_t,rgb_t>
       ^
In file included from deps/agg/src/agg_pixfmt_rgba.cpp:7:
In file included from /usr/local/include/boost/gil.hpp:12:
In file included from /usr/local/include/boost/gil/algorithm.hpp:11:
In file included from /usr/local/include/boost/gil/bit_aligned_pixel_iterator.hpp:11:
In file included from /usr/local/include/boost/gil/bit_aligned_pixel_reference.hpp:12:
/usr/local/include/boost/gil/pixel.hpp:104:32: error: type 'int' cannot be used prior to '::' because it has no members
        mp11::mp_size<typename Layout::color_space_t>::value
                               ^
deps/agg/src/agg_pixfmt_rgba.cpp:39:24: note: in instantiation of template class 'boost::gil::pixel<boost::gil::scoped_channel_value<float, boost::gil::float_point_zero<float>, boost::gil::float_point_one<float> >, int>' requested here
        hsv32f_pixel_t hsv_src,hsv_dst;
                       ^
In file included from deps/agg/src/agg_pixfmt_rgba.cpp:7:
In file included from /usr/local/include/boost/gil.hpp:12:
In file included from /usr/local/include/boost/gil/algorithm.hpp:11:
In file included from /usr/local/include/boost/gil/bit_aligned_pixel_iterator.hpp:11:
In file included from /usr/local/include/boost/gil/bit_aligned_pixel_reference.hpp:12:
/usr/local/include/boost/gil/pixel.hpp:113:36: error: type 'int' cannot be used prior to '::' because it has no members
            mp11::mp_size<typename Layout::color_space_t>::value
                                   ^
In file included from deps/agg/src/agg_pixfmt_rgba.cpp:7:
In file included from /usr/local/include/boost/gil.hpp:12:
In file included from /usr/local/include/boost/gil/algorithm.hpp:11:
In file included from /usr/local/include/boost/gil/bit_aligned_pixel_iterator.hpp:11:
In file included from /usr/local/include/boost/gil/bit_aligned_pixel_reference.hpp:12:
In file included from /usr/local/include/boost/gil/pixel.hpp:14:
/usr/local/include/boost/gil/color_base_algorithm.hpp:170:76: error: no member named 'layout_t' in 'boost::gil::pixel<boost::gil::scoped_channel_value<float, boost::gil::float_point_zero<float>, boost::gil::float_point_one<float> >, int>'
struct color_index_type : public detail::type_to_index<typename ColorBase::layout_t::color_space_t,Color> {};
                                                                ~~~~~~~~~~~^
/usr/local/include/boost/gil/color_base_algorithm.hpp:180:92: note: in instantiation of template class 'boost::gil::color_index_type<boost::gil::pixel<boost::gil::scoped_channel_value<float, boost::gil::float_point_zero<float>, boost::gil::float_point_one<float> >, int>, boost::gil::hsv_color_space::hue_t>' requested here
struct color_element_reference_type : public kth_semantic_element_reference_type<ColorBase,color_index_type<ColorBase,Color>::value> {};
                                                                                           ^
/usr/local/include/boost/gil/color_base_algorithm.hpp:190:10: note: in instantiation of template class 'boost::gil::color_element_reference_type<boost::gil::pixel<boost::gil::scoped_channel_value<float, boost::gil::float_point_zero<float>, boost::gil::float_point_one<float> >, int>, boost::gil::hsv_color_space::hue_t>' requested here
typename color_element_reference_type<ColorBase,Color>::type get_color(ColorBase& cb, Color=Color()) {
         ^
deps/agg/src/agg_pixfmt_rgba.cpp:42:9: note: while substituting deduced template arguments into function template 'get_color' [with ColorBase = boost::gil::pixel<boost::gil::scoped_channel_value<float, boost::gil::float_point_zero<float>, boost::gil::float_point_one<float> >, int>, Color = boost::gil::hsv_color_space::hue_t]
        get_color(hsv_dst,hue_t()) = get_color(hsv_src,hue_t());
        ^
In file included from deps/agg/src/agg_pixfmt_rgba.cpp:7:
In file included from /usr/local/include/boost/gil.hpp:12:
In file included from /usr/local/include/boost/gil/algorithm.hpp:11:
In file included from /usr/local/include/boost/gil/bit_aligned_pixel_iterator.hpp:11:
In file included from /usr/local/include/boost/gil/bit_aligned_pixel_reference.hpp:12:
In file included from /usr/local/include/boost/gil/pixel.hpp:14:
/usr/local/include/boost/gil/color_base_algorithm.hpp:185:139: error: no member named 'value' in 'boost::gil::color_index_type<boost::gil::pixel<boost::gil::scoped_channel_value<float, boost::gil::float_point_zero<float>, boost::gil::float_point_one<float> >, int>, boost::gil::hsv_color_space::hue_t>'
struct color_element_const_reference_type : public kth_semantic_element_const_reference_type<ColorBase,color_index_type<ColorBase,Color>::value> {};
                                                                                                       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
/usr/local/include/boost/gil/color_base_algorithm.hpp:197:10: note: in instantiation of template class 'boost::gil::color_element_const_reference_type<boost::gil::pixel<boost::gil::scoped_channel_value<float, boost::gil::float_point_zero<float>, boost::gil::float_point_one<float> >, int>, boost::gil::hsv_color_space::hue_t>' requested here
typename color_element_const_reference_type<ColorBase,Color>::type get_color(const ColorBase& cb, Color=Color()) {
         ^
deps/agg/src/agg_pixfmt_rgba.cpp:42:9: note: while substituting deduced template arguments into function template 'get_color' [with ColorBase = boost::gil::pixel<boost::gil::scoped_channel_value<float, boost::gil::float_point_zero<float>, boost::gil::float_point_one<float> >, int>, Color = boost::gil::hsv_color_space::hue_t]
        get_color(hsv_dst,hue_t()) = get_color(hsv_src,hue_t());
        ^
deps/agg/src/agg_pixfmt_rgba.cpp:42:9: error: no matching function for call to 'get_color'
        get_color(hsv_dst,hue_t()) = get_color(hsv_src,hue_t());
        ^~~~~~~~~
/usr/local/include/boost/gil/color_base_algorithm.hpp:190:62: note: candidate template ignored: substitution failure [with ColorBase = boost::gil::pixel<boost::gil::scoped_channel_value<float, boost::gil::float_point_zero<float>, boost::gil::float_point_one<float> >, int>, Color = boost::gil::hsv_color_space::hue_t]
typename color_element_reference_type<ColorBase,Color>::type get_color(ColorBase& cb, Color=Color()) {
                                                             ^
/usr/local/include/boost/gil/color_base_algorithm.hpp:197:68: note: candidate template ignored: substitution failure [with ColorBase = boost::gil::pixel<boost::gil::scoped_channel_value<float, boost::gil::float_point_zero<float>, boost::gil::float_point_one<float> >, int>, Color = boost::gil::hsv_color_space::hue_t]
typename color_element_const_reference_type<ColorBase,Color>::type get_color(const ColorBase& cb, Color=Color()) {
                                                                   ^
/usr/local/include/boost/gil/color_base_algorithm.hpp:170:76: error: no member named 'layout_t' in 'boost::gil::pixel<boost::gil::scoped_channel_value<float, boost::gil::float_point_zero<float>, boost::gil::float_point_one<float> >, int>'
struct color_index_type : public detail::type_to_index<typename ColorBase::layout_t::color_space_t,Color> {};
                                                                ~~~~~~~~~~~^
/usr/local/include/boost/gil/color_base_algorithm.hpp:180:92: note: in instantiation of template class 'boost::gil::color_index_type<boost::gil::pixel<boost::gil::scoped_channel_value<float, boost::gil::float_point_zero<float>, boost::gil::float_point_one<float> >, int>, boost::gil::hsv_color_space::saturation_t>' requested here
struct color_element_reference_type : public kth_semantic_element_reference_type<ColorBase,color_index_type<ColorBase,Color>::value> {};
                                                                                           ^
/usr/local/include/boost/gil/color_base_algorithm.hpp:190:10: note: in instantiation of template class 'boost::gil::color_element_reference_type<boost::gil::pixel<boost::gil::scoped_channel_value<float, boost::gil::float_point_zero<float>, boost::gil::float_point_one<float> >, int>, boost::gil::hsv_color_space::saturation_t>' requested here
typename color_element_reference_type<ColorBase,Color>::type get_color(ColorBase& cb, Color=Color()) {
         ^
deps/agg/src/agg_pixfmt_rgba.cpp:74:9: note: while substituting deduced template arguments into function template 'get_color' [with ColorBase = boost::gil::pixel<boost::gil::scoped_channel_value<float, boost::gil::float_point_zero<float>, boost::gil::float_point_one<float> >, int>, Color = boost::gil::hsv_color_space::saturation_t]
        get_color(hsv_dst,saturation_t()) = get_color(hsv_src,saturation_t());
        ^
In file included from deps/agg/src/agg_pixfmt_rgba.cpp:7:
In file included from /usr/local/include/boost/gil.hpp:12:
In file included from /usr/local/include/boost/gil/algorithm.hpp:11:
In file included from /usr/local/include/boost/gil/bit_aligned_pixel_iterator.hpp:11:
In file included from /usr/local/include/boost/gil/bit_aligned_pixel_reference.hpp:12:
In file included from /usr/local/include/boost/gil/pixel.hpp:14:
/usr/local/include/boost/gil/color_base_algorithm.hpp:185:139: error: no member named 'value' in 'boost::gil::color_index_type<boost::gil::pixel<boost::gil::scoped_channel_value<float, boost::gil::float_point_zero<float>, boost::gil::float_point_one<float> >, int>, boost::gil::hsv_color_space::saturation_t>'
struct color_element_const_reference_type : public kth_semantic_element_const_reference_type<ColorBase,color_index_type<ColorBase,Color>::value> {};
                                                                                                       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
/usr/local/include/boost/gil/color_base_algorithm.hpp:197:10: note: in instantiation of template class 'boost::gil::color_element_const_reference_type<boost::gil::pixel<boost::gil::scoped_channel_value<float, boost::gil::float_point_zero<float>, boost::gil::float_point_one<float> >, int>, boost::gil::hsv_color_space::saturation_t>' requested here
typename color_element_const_reference_type<ColorBase,Color>::type get_color(const ColorBase& cb, Color=Color()) {
         ^
deps/agg/src/agg_pixfmt_rgba.cpp:74:9: note: while substituting deduced template arguments into function template 'get_color' [with ColorBase = boost::gil::pixel<boost::gil::scoped_channel_value<float, boost::gil::float_point_zero<float>, boost::gil::float_point_one<float> >, int>, Color = boost::gil::hsv_color_space::saturation_t]
        get_color(hsv_dst,saturation_t()) = get_color(hsv_src,saturation_t());
        ^
deps/agg/src/agg_pixfmt_rgba.cpp:74:9: error: no matching function for call to 'get_color'
        get_color(hsv_dst,saturation_t()) = get_color(hsv_src,saturation_t());
        ^~~~~~~~~
/usr/local/include/boost/gil/color_base_algorithm.hpp:190:62: note: candidate template ignored: substitution failure [with ColorBase = boost::gil::pixel<boost::gil::scoped_channel_value<float, boost::gil::float_point_zero<float>, boost::gil::float_point_one<float> >, int>, Color = boost::gil::hsv_color_space::saturation_t]
typename color_element_reference_type<ColorBase,Color>::type get_color(ColorBase& cb, Color=Color()) {
                                                             ^
/usr/local/include/boost/gil/color_base_algorithm.hpp:197:68: note: candidate template ignored: substitution failure [with ColorBase = boost::gil::pixel<boost::gil::scoped_channel_value<float, boost::gil::float_point_zero<float>, boost::gil::float_point_one<float> >, int>, Color = boost::gil::hsv_color_space::saturation_t]
typename color_element_const_reference_type<ColorBase,Color>::type get_color(const ColorBase& cb, Color=Color()) {
                                                                   ^
deps/agg/src/agg_pixfmt_rgba.cpp:74:45: error: no matching function for call to 'get_color'
        get_color(hsv_dst,saturation_t()) = get_color(hsv_src,saturation_t());
                                            ^~~~~~~~~
/usr/local/include/boost/gil/color_base_algorithm.hpp:190:62: note: candidate template ignored: substitution failure [with ColorBase = boost::gil::pixel<boost::gil::scoped_channel_value<float, boost::gil::float_point_zero<float>, boost::gil::float_point_one<float> >, int>, Color = boost::gil::hsv_color_space::saturation_t]: no type named 'type' in 'boost::gil::color_element_reference_type<boost::gil::pixel<boost::gil::scoped_channel_value<float, boost::gil::float_point_zero<float>, boost::gil::float_point_one<float> >, int>, boost::gil::hsv_color_space::saturation_t>'
typename color_element_reference_type<ColorBase,Color>::type get_color(ColorBase& cb, Color=Color()) {
                                                        ~~~~ ^
/usr/local/include/boost/gil/color_base_algorithm.hpp:197:68: note: candidate template ignored: substitution failure [with ColorBase = boost::gil::pixel<boost::gil::scoped_channel_value<float, boost::gil::float_point_zero<float>, boost::gil::float_point_one<float> >, int>, Color = boost::gil::hsv_color_space::saturation_t]: no type named 'type' in 'boost::gil::color_element_const_reference_type<boost::gil::pixel<boost::gil::scoped_channel_value<float, boost::gil::float_point_zero<float>, boost::gil::float_point_one<float> >, int>, boost::gil::hsv_color_space::saturation_t>'
typename color_element_const_reference_type<ColorBase,Color>::type get_color(const ColorBase& cb, Color=Color()) {
                                                              ~~~~ ^
/usr/local/include/boost/gil/color_base_algorithm.hpp:170:76: error: no member named 'layout_t' in 'boost::gil::pixel<boost::gil::scoped_channel_value<float, boost::gil::float_point_zero<float>, boost::gil::float_point_one<float> >, int>'
struct color_index_type : public detail::type_to_index<typename ColorBase::layout_t::color_space_t,Color> {};
                                                                ~~~~~~~~~~~^
/usr/local/include/boost/gil/color_base_algorithm.hpp:180:92: note: in instantiation of template class 'boost::gil::color_index_type<boost::gil::pixel<boost::gil::scoped_channel_value<float, boost::gil::float_point_zero<float>, boost::gil::float_point_one<float> >, int>, boost::gil::hsl_color_space::hue_t>' requested here
struct color_element_reference_type : public kth_semantic_element_reference_type<ColorBase,color_index_type<ColorBase,Color>::value> {};
                                                                                           ^
/usr/local/include/boost/gil/color_base_algorithm.hpp:190:10: note: in instantiation of template class 'boost::gil::color_element_reference_type<boost::gil::pixel<boost::gil::scoped_channel_value<float, boost::gil::float_point_zero<float>, boost::gil::float_point_one<float> >, int>, boost::gil::hsl_color_space::hue_t>' requested here
typename color_element_reference_type<ColorBase,Color>::type get_color(ColorBase& cb, Color=Color()) {
         ^
deps/agg/src/agg_pixfmt_rgba.cpp:106:9: note: while substituting deduced template arguments into function template 'get_color' [with ColorBase = boost::gil::pixel<boost::gil::scoped_channel_value<float, boost::gil::float_point_zero<float>, boost::gil::float_point_one<float> >, int>, Color = boost::gil::hsl_color_space::hue_t]
        get_color(hsl_dst,hue_t()) = get_color(hsl_src,hue_t());
        ^
In file included from deps/agg/src/agg_pixfmt_rgba.cpp:7:
In file included from /usr/local/include/boost/gil.hpp:12:
In file included from /usr/local/include/boost/gil/algorithm.hpp:11:
In file included from /usr/local/include/boost/gil/bit_aligned_pixel_iterator.hpp:11:
In file included from /usr/local/include/boost/gil/bit_aligned_pixel_reference.hpp:12:
In file included from /usr/local/include/boost/gil/pixel.hpp:14:
/usr/local/include/boost/gil/color_base_algorithm.hpp:185:139: error: no member named 'value' in 'boost::gil::color_index_type<boost::gil::pixel<boost::gil::scoped_channel_value<float, boost::gil::float_point_zero<float>, boost::gil::float_point_one<float> >, int>, boost::gil::hsl_color_space::hue_t>'
struct color_element_const_reference_type : public kth_semantic_element_const_reference_type<ColorBase,color_index_type<ColorBase,Color>::value> {};
                                                                                                       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
/usr/local/include/boost/gil/color_base_algorithm.hpp:197:10: note: in instantiation of template class 'boost::gil::color_element_const_reference_type<boost::gil::pixel<boost::gil::scoped_channel_value<float, boost::gil::float_point_zero<float>, boost::gil::float_point_one<float> >, int>, boost::gil::hsl_color_space::hue_t>' requested here
typename color_element_const_reference_type<ColorBase,Color>::type get_color(const ColorBase& cb, Color=Color()) {
         ^
deps/agg/src/agg_pixfmt_rgba.cpp:106:9: note: while substituting deduced template arguments into function template 'get_color' [with ColorBase = boost::gil::pixel<boost::gil::scoped_channel_value<float, boost::gil::float_point_zero<float>, boost::gil::float_point_one<float> >, int>, Color = boost::gil::hsl_color_space::hue_t]
        get_color(hsl_dst,hue_t()) = get_color(hsl_src,hue_t());
        ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
```

**Comment by artemp** [comment]
@jbeich - ha, I'm failing to catch up with new boost releases :)  thanks!

**Comment by artemp** [comment]
@jbeich - ok, 1_72 hasn't been released, yet, but thanks for heads up.

**Comment by gen2** [comment]
I run into a different issue:

```
x86_64-pc-linux-gnu-g++ -o src/json/mapnik_geometry_to_geojson.o -c -std=c++11 -DU_USING_ICU_NAMESPACE=0 -fvisibility=hidden -fvisibility-inlines-hidden -Wall -pthread -ftemplate-depth-300 -Wsign-compare -Wshadow -O2 -pipe -fPIC -DMAPNIK_MEMORY_MAPPED_FILE -DMAPNIK_HAS_DLCFN -DBIGINT -DBOOST_REGEX_HAS_ICU -DHAVE_JPEG -DMAPNIK_USE_PROJ4 -DHAVE_PNG -DHAVE_WEBP -DHAVE_TIFF -DLINUX -DMAPNIK_THREADSAFE -DBOOST_SPIRIT_NO_PREDEFINED_TERMINALS=1 -DBOOST_PHOENIX_NO_PREDEFINED_TERMINALS=1 -DBOOST_SPIRIT_USE_PHOENIX_V3=1 -DNDEBUG -Ideps -Ideps/mapbox/variant/include -Ideps/agg/include -Iinclude -I/usr/include -I/usr/include/freetype2 -I/usr/include/libxml2 -I/usr/include/gdal -I/usr/include/postgresql-12 src/json/mapnik_geometry_to_geojson.cpp
In file included from /usr/include/boost/type_traits/is_convertible.hpp:20,
                 from /usr/include/boost/proto/matches.hpp:38,
                 from /usr/include/boost/phoenix/core/domain.hpp:12,
                 from /usr/include/boost/phoenix/core/actor.hpp:18,
                 from /usr/include/boost/phoenix/core.hpp:12,
                 from /usr/include/boost/spirit/include/phoenix_core.hpp:11,
                 from /usr/include/boost/spirit/home/support/terminal.hpp:17,
                 from /usr/include/boost/spirit/home/support/common_terminals.hpp:15,
                 from /usr/include/boost/spirit/home/karma/char/char.hpp:14,
                 from /usr/include/boost/spirit/home/karma/char.hpp:13,
                 from /usr/include/boost/spirit/home/karma.hpp:13,
                 from /usr/include/boost/spirit/include/karma.hpp:16,
                 from include/mapnik/json/geometry_generator_grammar.hpp:33,
                 from src/json/mapnik_geometry_to_geojson.cpp:25:
/usr/include/boost/spirit/home/support/attributes.hpp: In instantiation of 'struct boost::spirit::traits::transform_attribute<const mapnik::geometry::geometry<double>, const mapnik::geometry::geometry<double>&, boost::spirit::karma::domain, void>':
/usr/include/boost/spirit/home/karma/nonterminal/rule.hpp:293:42:   required from 'bool boost::spirit::karma::rule<OutputIterator, T1, T2, T3, T4>::generate(boost::spirit::karma::rule<OutputIterator, T1, T2, T3, T4>::output_iterator&, Context&, const Delimiter&, const Attribute&) const [with Context = boost::spirit::context<boost::fusion::cons<const mapnik::geometry::geometry<double>&, boost::fusion::nil_>, boost::spirit::locals<> >; Delimiter = boost::spirit::unused_type; Attribute = mapnik::geometry::geometry<double>; OutputIterator = std::back_insert_iterator<std::__cxx11::basic_string<char> >; T1 = const mapnik::geometry::geometry<double>&(); T2 = boost::spirit::unused_type; T3 = boost::spirit::unused_type; T4 = boost::spirit::unused_type; boost::spirit::karma::rule<OutputIterator, T1, T2, T3, T4>::output_iterator = boost::spirit::karma::detail::output_iterator<std::back_insert_iterator<std::__cxx11::basic_string<char> >, mpl_::int_<15>, boost::spirit::unused_type>]'
/usr/include/boost/spirit/home/karma/reference.hpp:46:65:   required from 'bool boost::spirit::karma::reference<Subject>::generate(OutputIterator&, Context&, const Delimiter&, const Attribute&) const [with OutputIterator = boost::spirit::karma::detail::output_iterator<std::back_insert_iterator<std::__cxx11::basic_string<char> >, mpl_::int_<15>, boost::spirit::unused_type>; Context = boost::spirit::context<boost::fusion::cons<const mapnik::geometry::geometry<double>&, boost::fusion::nil_>, boost::spirit::locals<> >; Delimiter = boost::spirit::unused_type; Attribute = mapnik::geometry::geometry<double>; Subject = const boost::spirit::karma::rule<std::back_insert_iterator<std::__cxx11::basic_string<char> >, const mapnik::geometry::geometry<double>&(), boost::spirit::unused_type, boost::spirit::unused_type, boost::spirit::unused_type>]'
/usr/include/boost/spirit/home/karma/generate.hpp:69:81:   required from 'bool boost::spirit::karma::generate(boost::spirit::karma::detail::output_iterator<OutputIterator, Derived>&, const Expr&, const Attr&) [with OutputIterator = std::back_insert_iterator<std::__cxx11::basic_string<char> >; Properties = mpl_::int_<15>; Expr = mapnik::json::geometry_generator_grammar<std::back_insert_iterator<std::__cxx11::basic_string<char> >, mapnik::geometry::geometry<double> >; Attr = mapnik::geometry::geometry<double>]'
/usr/include/boost/spirit/home/karma/generate.hpp:91:31:   required from 'bool boost::spirit::karma::generate(OutputIterator&, const Expr&, const Attr&) [with OutputIterator = std::back_insert_iterator<std::__cxx11::basic_string<char> >; Expr = mapnik::json::geometry_generator_grammar<std::back_insert_iterator<std::__cxx11::basic_string<char> >, mapnik::geometry::geometry<double> >; Attr = mapnik::geometry::geometry<double>]'
src/json/mapnik_geometry_to_geojson.cpp:34:62:   required from here
/usr/include/boost/spirit/home/support/attributes.hpp:963:9: error: static assertion failed: Transformed cannot be a reference type
         BOOST_STATIC_ASSERT_MSG(!is_reference<Transformed>::value,
         ^~~~~~~~~~~~~~~~~~~~~~~

```

boost is on version 1.71.0 (current stable on gentoo linux)

Should i open a separate issue on this?

**Comment by artemp** [comment]
@gen2 -yes, pls ^ separate issue would be great, thanks

**Comment by gen2** [comment]
> @gen2 -yes, pls ^ separate issue would be great, thanks

[it's here](https://github.com/mapnik/mapnik/issues/4098)

**Comment by jbeich** [comment]
Boost 1.72 is [released](https://lists.boost.org/Archives/boost/2019/12/247795.php) and [landed](https://github.com/freebsd/freebsd-ports/commit/837be5f8be2597d837d1373625cb283eb9865e7b) in FreeBSD Ports. Given Mapnik never supports **current** Boost version maybe add an option to build against bundled/internal Boost version that Mapnik does support.

**Comment by artemp** [comment]
@jbeich - FYI, latest master works for me with BOOST 1_72. Give it a go and report if any issues, thanks.

**Comment by jbeich** [comment]
I confirm, [builds fine](https://github.com/mapnik/mapnik/files/3962045/mapnik-3.0.12.1078.log) after #4103.

**Comment by am2222** [comment]
can this bug be related to this error?
https://github.com/mapnik/python-mapnik/issues/219
